device/thunder_ripple_sdk/src/client/device_operator.rs

```diff
@@ -174,11 +174,14 @@ impl DeviceResponseMessage {
         json_resp: &JsonRpcApiResponse,
         sub_id: Option<String>,
     ) -> Option<DeviceResponseMessage> {
+        let json_value = serde_json::to_value(json_resp).ok();
         let mut device_response_msg = None;
         if let Some(res) = &json_resp.result {
             device_response_msg = Some(DeviceResponseMessage::new(res.clone(), sub_id));
-        } else if let Some(er) = &json_resp.error {
-            device_response_msg = Some(DeviceResponseMessage::new(er.clone(), sub_id));
+        } else if let Some(_er) = &json_resp.error {
+            if let Some(val) = json_value {
```

Comment:
No need to create `json_value` until here so you should just do something like:

`if let Ok(json_value) = serde_json::to_value(json_resp) {`

...but I'm not quite following why in the error case you're setting the `DeviceResponseMessage` equal to the entire `JsonRpcApiResponse`, won't the `DeviceResponseMessage` already be wrapped? In other words why are the error and non-error cases so different? Maybe @Vinodsathyaseelan can answer that?